shell/browser/notifications/mac/cocoa_notification.mm

```diff
@@ -26,101 +26,185 @@
     : Notification(delegate, presenter) {}
 
 CocoaNotification::~CocoaNotification() {
-  if (notification_)
-    [NSUserNotificationCenter.defaultUserNotificationCenter
-        removeDeliveredNotification:notification_];
+  if (notification_request_)
+    [[UNUserNotificationCenter currentNotificationCenter]
+        removeDeliveredNotificationsWithIdentifiers:@[
+          notification_request_.identifier
+        ]];
 }
 
 void CocoaNotification::Show(const NotificationOptions& options) {
-  notification_ = [[NSUserNotification alloc] init];
+  UNMutableNotificationContent* content =
+      [[UNMutableNotificationContent alloc] init];
 
-  NSString* identifier =
-      [NSString stringWithFormat:@"%@:notification:%@",
-                                 [[NSBundle mainBundle] bundleIdentifier],
-                                 [[NSUUID UUID] UUIDString]];
-
-  [notification_ setTitle:base::SysUTF16ToNSString(options.title)];
-  [notification_ setSubtitle:base::SysUTF16ToNSString(options.subtitle)];
-  [notification_ setInformativeText:base::SysUTF16ToNSString(options.msg)];
-  [notification_ setIdentifier:identifier];
-
-  if (electron::debug_notifications) {
-    LOG(INFO) << "Notification created (" << [identifier UTF8String] << ")";
-  }
+  content.title = base::SysUTF16ToNSString(options.title);
+  content.subtitle = base::SysUTF16ToNSString(options.subtitle);
+  content.body = base::SysUTF16ToNSString(options.msg);
 
   if (!options.icon.drawsNothing()) {
     NSImage* image = skia::SkBitmapToNSImage(options.icon);
-    [notification_ setContentImage:image];
+    NSData* imageData =
```

Comment:
Instead of this custom logic, can we use `NotificationImageRetainer` from `//chrome` (either copy or include). It was built to do this but also supports automatic cleanup of the temporary images.

https://source.chromium.org/chromium/chromium/src/+/main:chrome/common/notifications/notification_image_retainer.h

Reply:
Adopted this pattern in the latest update, thank you!